tests/scene/test_option_button.h

```diff
@@ -0,0 +1,142 @@
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_OPTION_BUTTON_H
+#define TEST_OPTION_BUTTON_H
+
+#include "scene/gui/option_button.h"
+
+#include "tests/test_macros.h"
+
+namespace TestOptionButton {
+
+TEST_CASE("[SceneTree][OptionButton] Initialization") {
+	OptionButton *test_opt = memnew(OptionButton);
+
+	SUBCASE("There should be no options right after initialization") {
+		CHECK_FALSE(test_opt->has_selectable_items());
+		CHECK(test_opt->get_item_count() == 0);
+	}
+
+	memdelete(test_opt);
+}
+
+TEST_CASE("[SceneTree][OptionButton] Single item") {
+	OptionButton *test_opt = memnew(OptionButton);
+
+	SUBCASE("There should a single item after after adding one") {
+		test_opt->add_item("single", 1013);
+
+		CHECK(test_opt->has_selectable_items());
+		CHECK(test_opt->get_item_count() == 1);
+		CHECK(test_opt->get_item_index(1013) == 0);
+		CHECK(test_opt->get_item_id(0) == 1013);
+
+		test_opt->remove_item(0);
+
+		CHECK_FALSE(test_opt->has_selectable_items());
+		CHECK(test_opt->get_item_count() == 0);
+	}
+
+	SUBCASE("There should a single item after after adding an icon") {
+		Ref<Texture2D> *test_icon = memnew(Ref<Texture2D>);
+		test_opt->add_icon_item(*test_icon, "icon", 345);
+
+		CHECK(test_opt->has_selectable_items());
+		CHECK(test_opt->get_item_count() == 1);
+		CHECK(test_opt->get_item_index(345) == 0);
+		CHECK(test_opt->get_item_id(0) == 345);
+
+		test_opt->remove_item(0);
+
+		CHECK_FALSE(test_opt->has_selectable_items());
+		CHECK(test_opt->get_item_count() == 0);
+
+		memdelete(test_icon);
+	}
+
+	memdelete(test_opt);
+}
+
+TEST_CASE("[SceneTree][OptionButton] Complex structure") {
+	OptionButton *test_opt = memnew(OptionButton);
+
+	SUBCASE("Creating a complex structure and checking getters") {
+		// regular item at index 0
```

Comment:
```suggestion
		// Regular item at index 0.
```
Per the [comment style](https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html#comment-style-guide). For all

Reply:
Thank you.

Reply:
I forgot the _periods_. A moment, I will action this fully! 😌

Reply:
(you missed the period)